tests/Unit/Helpers/Evolution/DiscontinuousGalerkin/BoundaryCorrections.hpp

```diff
@@ -315,4 +316,211 @@ void test_boundary_correction_conservation(
+                         interior_package_data, exterior_package_data,
+                         dg_formulation);
+
+  // Call python implementation of dg_boundary_terms
+  function_name_index = 0;
+  tmpl::for_each<VariablesTags>([&boundary_corrections, &dg_formulation,
+                                 epsilon, &exterior_package_data,
+                                 &function_name_index, &interior_package_data,
+                                 &python_dg_boundary_terms_functions,
+                                 &python_module](auto package_data_tag_v) {
+    INFO("Testing boundary terms");
+    using boundary_correction_tag =
+        tmpl::type_from<decltype(package_data_tag_v)>;
+    using ResultType = typename boundary_correction_tag::type;
+    try {
+      CAPTURE(python_module);
+      const std::string& python_function =
+          gsl::at(python_dg_boundary_terms_functions, function_name_index);
+      CAPTURE(python_function);
+      // Make explicitly depend on tag type to avoid type deduction issues with
+      // GCC7
+      const typename boundary_correction_tag::type python_result =
+          pypp::call<ResultType>(
+              python_module, python_function,
+              get<DgPackageDataTags>(interior_package_data)...,
+              get<DgPackageDataTags>(exterior_package_data)...,
+              dg_formulation == ::dg::Formulation::StrongInertial);
+      CHECK_ITERABLE_CUSTOM_APPROX(
+          get<boundary_correction_tag>(boundary_corrections), python_result,
+          Approx::custom().epsilon(epsilon).scale(1.0));
+    } catch (const std::exception& e) {
+      INFO("On line " << __LINE__ << " Python call to "
+                      << gsl::at(python_dg_boundary_terms_functions,
+                                 function_name_index)
+                      << " in module " << python_module
+                      << " failed: " << e.what());
+      REQUIRE(false);
+    }
+    ++function_name_index;
+  });
+}
+}  // namespace detail
+
+/*!
+ * \ingroup TestingFrameworkGroup
+ * \brief Tests that the `dg_package_data` and `dg_boundary_terms` functions
+ * agree with the python implementation.
+ *
+ * The variables are filled with random numbers between zero and one before
+ * being passed to the implementations. If in the future we need support for
+ * negative numbers we can add the ability to specify a single range for all
+ * random numbers or each individually.
+ *
+ * The `pypp::SetupLocalPythonEnvironment` must be created before the
+ * `test_boundary_correction_with_python` can be called.
+ *
+ * The `dg_formulation` is passed as a bool `use_strong_form` to the python
+ * functions since we don't want to rely on python bindings for the enum.
+ *
+ * You can convert custom types using the `ConversionClassList` template
+ * parameter, which is then passed to `pypp::call()`. This allows you to, e.g.,
+ * convert an equation of state into an array locally in a test file.
+ */
```

Comment:
It would be helpful to add notes about the assumed relationship between python functions and the package functions tag list, and the `System:variables_tag` tag list - In particular, you require that there be a python function for each packaged field, and one for each tag in the system variables. The interface of the python function should also be specified -- the order of arguments and relationships to the tag collections where the `Variables` need to be pack-expanded.